src/2d/shallow/landspill/darcy_weisbach/darcy_weisbach_abstract_module.f90

```diff
@@ -0,0 +1,211 @@
+!
+! darcy_weisbach_abstract_module.f90
+! Copyright (C) 2018 Pi-Yueh Chuang <pychuang@gwu.edu>
+!
+! Distributed under terms of the MIT license.
+!
+
```

Comment:
It's not technically needed but I suggest putting `implicit none` in every subroutine just to remind the subsequent user that they need to define everything.